source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
 
-It is important to use ``addBehaviour`` within the same case class.
+.. 1. Since ``House`` is a component of the Network, you have to import the Network class.
 
-.. code-block:: scala
+.. .. code-block:: scala
 
-  addBehaviour(watchMovie)
+..   import com.bharatsim.engine.models.Network
 
-Saving your output
-^^^^^^^^^^^^^^^^^^
+.. 2. The ``House`` class is a `case class <https://docs.scala-lang.org/tour/case-classes.html>`_ and it extends the framework defined Network class.
 
-Suppose you wanted your output to give you the numbers of susceptible, infected and recovered people at every time step. You can then write the following:
+.. .. code-block:: scala
 
-.. code-block:: scala
+..   case class House extends Network
 
-  import com.bharatsim.engine.Context
-  import com.bharatsim.engine.graph.patternMatcher.MatchCondition._
-  import com.bharatsim.engine.listeners.CSVSpecs
-  import com.bharatsim.examples.epidemiology.SIR.InfectionStatus.{Infected, Removed, Susceptible}
-
-  class SIROutputSpec(context: Context) extends CSVSpecs {
-    override def getHeaders: List[String] =
-      List(
-        "Step",
-        "Susceptible",
-        "Infected",
-        "Removed"
-      )
-
-    override def getRows(): List[List[Any]] = {
-      val graphProvider = context.graphProvider
-      val label = "Person"
-      val row = List(
-        context.getCurrentStep,
-        graphProvider.fetchCount(label, "infectionState" equ Susceptible),
-        graphProvider.fetchCount(label, "infectionState" equ Infected),
-        graphProvider.fetchCount(label, "infectionState" equ Removed)
-      )
-      List(row)
-    }
-  }
+.. 3. However, each house in the Network requires a unique house id which is given as an argument to the ``House`` class. This house id is an ‘attribute’ of the corresponding house.
 
-* The first column (Step) stores the current time step, obtained using the ``context.getCurrentStep`` function
-* The next 3 columns store the number of Susceptible, Infected and Removed people respectively, by fetching the total number of ``Person`` nodes on the graph with the appropriate appropriate `infection status <#>`_.
+.. .. code-block:: scala
 
-Now we simply have to register it in the simulation. Note that the following code snippet should be located inside ``simulation.defineSimulation`` in the main function:
+..   case class House(id: Long) extends Network
 
-.. code-block:: scala
+.. .. note:: Long is just a datatype of Scala.
 
-  SimulationListenerRegistry.register(
-    new CsvOutputGenerator("src/main/resources/output.csv", new SIROutputSpec(context))
-      )
+.. 4. You can define the relationship of the case-class with the agent by using the framework defined ``addRelation`` function within the class definition. A house houses an agent, so the relation is simply given by the string, “HOUSES”. These relations are defined in the Main class and are user-definable.

```

Comment:
Which class? Say something like:

within the definition of any class that extends the `Node` class. These include any extensions of the `Agent` and `Network` classes which themselves extend the `Node` class.

Or something like that.